queue/retry.go

```diff
@@ -178,6 +204,13 @@ func (rh *basicRetryHandler) waitForJob(ctx context.Context) {
 				}
 			}
 
+			rh.mu.Lock()
```

Comment:
Should L177 also acquire the `rh` mutex when putting the job in the `pending` map?

Reply:
Same comment about temporary stats.